agent/functional_tests/tests/functionaltests_unix_test.go

```diff
@@ -618,3 +618,45 @@ func TestNetworkModeBridge(t *testing.T) {
 	err := networkModeTest(t, agent, "bridge")
 	require.NoError(t, err, "Networking mode bridge testing failed")
 }
+
+// TestCustomAttributesWithMaxOptions tests the ECS_INSTANCE_ATTRIBUTES
+// upon agent registration with maximum number of supported key, value pairs
+func TestCustomAttributesWithMaxOptions(t *testing.T) {
```

Comment:
Are you following up immediately with Windows?  This will continue to be a gap in our coverage until we have a Windows test.

The test as written should work on Windows too.  The only thing you'd have to do is modify the Windows version of `StartAgent` to use the new `ECS_FTEST_INSTANCE_ATTRIBUTES` environment variable you specified.